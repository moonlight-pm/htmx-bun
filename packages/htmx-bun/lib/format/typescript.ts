import { format } from "prettier";

export async function formatTypeScript(code: string): Promise<string> {
    return format(code, {
        parser: "typescript",
        semi: true,
        singleQuote: false,
        trailingComma: "all",
        arrowParens: "always",
    });
}
