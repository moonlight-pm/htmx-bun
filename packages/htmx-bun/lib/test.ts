import { afterAll } from "bun:test";
import { mkdtempSync, rmdirSync } from "node:fs";
import { tmpdir } from "node:os";

export function makeTemporaryDirectory() {
    const dir = mkdtempSync(`${tmpdir()}/htmx-bun-`);
    afterAll(() => {
        rmdirSync(dir, { recursive: true });
    });
    return dir;
}
