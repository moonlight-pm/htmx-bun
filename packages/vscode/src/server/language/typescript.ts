import { CodeMapping, VirtualCode } from "@volar/language-core";
import { IScriptSnapshot } from "typescript";

export class TypeScriptVirtualCode implements VirtualCode {
    id: string;
    languageId = "typescript";
    snapshot: IScriptSnapshot;
    mappings: CodeMapping[];
    embeddedCodes = [];

    constructor(parentId: string, offset: number, text: string) {
        this.id = `${parentId}:typescript`;
        this.snapshot = {
            getText: (start, end) => text.slice(start, end),
            getLength: () => text.length,
            getChangeRange: () => undefined,
        };
        this.mappings = [
            {
                sourceOffsets: [offset],
                generatedOffsets: [0],
                lengths: [text.length],
                data: {
                    completion: true,
                    format: false,
                    navigation: true,
                    semantic: true,
                    structure: true,
                    verification: true,
                },
            },
        ];
    }
}
