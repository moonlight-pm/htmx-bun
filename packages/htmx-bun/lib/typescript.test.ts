import { expect, test } from "bun:test";
import { formatTypeScript } from "./typescript";

test("formatTypeScript", () => {
    const ugly = `
      const a    =
      1;
      const b = "foo"

      function  c   (  ) { return null
        ;}

    `;
    const pretty = formatTypeScript(ugly);
    expect(pretty).toMatchSnapshot();
});
