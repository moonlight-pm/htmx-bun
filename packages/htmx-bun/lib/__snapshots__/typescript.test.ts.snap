// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`formatTypeScript 1`] = `
"const a = 1;
const b = "foo";
function c() {
    return null;
}
"
`;
